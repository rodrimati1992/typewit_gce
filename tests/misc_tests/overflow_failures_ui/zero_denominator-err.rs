const N: u32 = 10;

fn rem_zero() {
    typewit_gce::gce_int_eq!(N % 0, 0u32);
    typewit_gce::gce_int_eq!((N + 4) % (N - N), 0u32);
}

fn div_zero() {
    typewit_gce::gce_int_eq!(N / 0, 0u32);
    typewit_gce::gce_int_eq!((N - 4) / (N - N), 0u32);
}



fn main(){}


