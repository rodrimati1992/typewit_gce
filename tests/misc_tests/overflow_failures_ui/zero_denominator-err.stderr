error: attempted to divide `N` by zero
 --> tests/misc_tests/overflow_failures_ui/zero_denominator-err.rs:4:5
  |
4 |     typewit_gce::gce_int_eq!(N % 0, 0u32);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error: attempted to divide `4 + N` by zero
 --> tests/misc_tests/overflow_failures_ui/zero_denominator-err.rs:5:5
  |
5 |     typewit_gce::gce_int_eq!((N + 4) % (N - N), 0u32);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error: attempted to divide `N` by zero
 --> tests/misc_tests/overflow_failures_ui/zero_denominator-err.rs:9:5
  |
9 |     typewit_gce::gce_int_eq!(N / 0, 0u32);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error: attempted to divide `- 4 + N` by zero
  --> tests/misc_tests/overflow_failures_ui/zero_denominator-err.rs:10:5
   |
10 |     typewit_gce::gce_int_eq!((N - 4) / (N - N), 0u32);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: attempt to calculate the remainder of `10_u32` with a divisor of zero
 --> tests/misc_tests/overflow_failures_ui/zero_denominator-err.rs:4:30
  |
4 |     typewit_gce::gce_int_eq!(N % 0, 0u32);
  |                              ^^^^^ evaluation of `rem_zero::{constant#0}` failed here

error[E0080]: attempt to calculate the remainder of `14_u32` with a divisor of zero
 --> tests/misc_tests/overflow_failures_ui/zero_denominator-err.rs:5:30
  |
5 |     typewit_gce::gce_int_eq!((N + 4) % (N - N), 0u32);
  |                              ^^^^^^^^^^^^^^^^^ evaluation of `rem_zero::{constant#2}` failed here

error[E0080]: attempt to divide `10_u32` by zero
 --> tests/misc_tests/overflow_failures_ui/zero_denominator-err.rs:9:30
  |
9 |     typewit_gce::gce_int_eq!(N / 0, 0u32);
  |                              ^^^^^ evaluation of `div_zero::{constant#0}` failed here

error[E0080]: attempt to divide `6_u32` by zero
  --> tests/misc_tests/overflow_failures_ui/zero_denominator-err.rs:10:30
   |
10 |     typewit_gce::gce_int_eq!((N - 4) / (N - N), 0u32);
   |                              ^^^^^^^^^^^^^^^^^ evaluation of `div_zero::{constant#2}` failed here
